onnxruntime/python/onnxruntime_collect_build_info.py

```diff
@@ -0,0 +1,89 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import warnings
+import ctypes
+import sys
+
+
+def find_cudart_versions(build_env=False, build_cuda_version=None):
+    # ctypes.CDLL and ctypes.util.find_library load the latest installed library.
+    # it may not the the library that would be loaded by onnxruntime.
+    # for example, in an environment with Cuda 11.1 and subsequently
+    # conda cudatoolkit 10.2.89 installed. ctypes will find cudart 10.2. however,
+    # onnxruntime built with Cuda 11.1 will find and load cudart for Cuda 11.1.
+    # for the above reason, we need find all versions in the environment and
+    # only give warnings if the expected cuda version is not found.
+    # in onnxruntime build environment, we expected only one Cuda version.
+    if not sys.platform.startswith('linux'):
+        warnings.warn('find_cudart_versions only works on Linux')
+        return None
+
+    cudart_possible_versions = {None, build_cuda_version}
+
+    def get_cudart_version(find_cudart_version=None):
+        cudart_lib_filename = 'libcudart.so'
+        if find_cudart_version:
+            cudart_lib_filename = cudart_lib_filename + '.' + find_cudart_version
+
+        try:
+            cudart = ctypes.CDLL(cudart_lib_filename)
+            cudart.cudaRuntimeGetVersion.restype = int
+            cudart.cudaRuntimeGetVersion.argtypes = [ctypes.POINTER(ctypes.c_int)]
+            version = ctypes.c_int()
+            status = cudart.cudaRuntimeGetVersion(ctypes.byref(version))
+            if status != 0:
+                return None
+        except: # noqa
+            return None
+
+        return version.value
+
+    # use set to avoid duplications
+    cudart_found_versions = {
+        get_cudart_version(cudart_version) for cudart_version in cudart_possible_versions}
+
+    # convert to list and remove None
+    return [ver for ver in cudart_found_versions if ver]
+
+
+def find_cudnn_supported_cuda_versions(build_env=False):
+    # comments in get_cudart_version apply here
+    if not sys.platform.startswith('linux'):
+        warnings.warn('find_cudnn_versions only works on Linux')
```

Comment:
Do this and the other warning above mean we'll log two warnings every time somebody tries to import `onnxruntime` or `torch_ort` on Windows, even when everything is set up correctly? (Also, shouldn't we be exiting early here?)

I'd prefer to avoid this. Perhaps we can warn only during setup?

Reply:
validate_build_package_info() is only called the first time:
from onnxruntime.capi import onnxruntime_validation
is invoked. Later invocation has no effect. 
In addition, validate_build_package_info() is coded in a way that it will post message only when the current ort installation is for ortmodule (e.g. training and gpu) AND when there is something we know that is wrong (either it is a missing/wrong Cuda RT version or the build is missing build_and_package_info).